internal/common/function/function.go

```diff
@@ -139,6 +140,51 @@ func GenerateVeleroBackupName(namespace, nabName string) string {
 	return veleroBackupName
 }
 
+// GenerateVeleroBackupNameWithUUID generates a Velero backup name based on the provided namespace and NonAdminBackup name.
+// It includes a UUID suffix. If the name exceeds the maximum length, it truncates nabName first, then namespace.
+func GenerateVeleroBackupNameWithUUID(namespace, nabName string) string {
+	// Generate UUID suffix
+	uuidSuffix := uuid.New().String()
+
+	// Build the initial backup name based on the presence of namespace and nabName
+	veleroBackupName := uuidSuffix
+	if len(nabName) > 0 {
+		veleroBackupName = nabName + constant.BackupNameDelimiter + veleroBackupName
+	}
+	if len(namespace) > 0 {
```

Comment:
Maybe we can drop using namespace in VeleroBackupName. It will anyways be part of metadata/spec ?

Reply:
I think it's useful to quickly filter the namespace from which Backup was created. So if we have multiple users with admin rights that have separate namesapces it allows to quickly sort/identify which Backups are from which ns. Of course this will be also available as additional info via annotation or label, but here is just another step to not have one list of unreadable by human backup names (uuid's only as example).

Reply:
If we are using a constant prefix in NAB names, this function needs to change

way it is, we call it and then check if result is valid; if not, we will retry

but on retry, only thing that will change is name suffix, which has the UUID

Correct way should be to create a function that generates the prefix (in this case `<nab-name>-<nab-namespace>-`, truncated) if needed

then, inside retry loop, we create UUID and add result of previous function and try to create velero backup. If it fails, we just retry with new UUID, prefix was already created and will not change

Reply:
It's essentially the same, just a different approach to writing the function. I prefer to encapsulate everything in a single function to generate the full name. If we go with your approach, we'd need to:

1. Create function to get `<nab-name>-<nab-namespace>-` with additional parameter of max length (we do not know if UUID will always have static length)
2. Make logic to merge above with generated UUID
3. Make logic to retry if the name exists inside reconcile

With current implementation we skip entirely point 2 making it easier within reconcile.

Reply:
from this https://pkg.go.dev/github.com/google/uuid#Parse it is always 36 chars length

Reply:
Correct if we take RFC 4122, then is always 36 but really it should not be split between two functions to create string.